### PR TITLE
[VIM-7179] Fixes for UserItem and updates for Folder models

### PR DESCRIPTION
**Pull request by asongvimeo**
#### Ticket

[VIM-7179](https://vimean.atlassian.net/browse/VIM-7179)

- *Note: this is required for Vimeo staff only. If not applicable to your PR, use "N/A".*

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

- UserItemType not mapping correctly on UserItem
- Folder 

#### Implementation Summary

- UserItem: Added missing mapping logic for the UserItemType property.
- Folder: Added missing logic for converting the date string to date types.

#### Reviewer Tips

- n/a

#### How to Test

- n/a

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">SwiftLint not installed, download from https://github.com/realm/SwiftLint</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 117 tests, with 0 failures (0 unexpected) in 0.675 (0.733) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/1213)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: Sources/Shared/Models/Folder.swift [hunk 1]
```diff
@@ -29,13 +29,16 @@ import Foundation
 public class Folder: VIMModelObject, ConnectionsProviding, ConnectionsParsing {
     
     /// The created time for the `Folder`
-    @objc public private(set) var createdTime: NSDate?
+    @objc public private(set) var createdTimeString: String?
```
Comment (bryantjustin) [inline Sources/Shared/Models/Folder.swift]: Not necessary. Just use `createdTime` and type it as `Date`.

File: Sources/Shared/Models/Folder.swift [hunk 2]
```diff
@@ -75,22 +78,29 @@ public class Folder: VIMModelObject, ConnectionsProviding, ConnectionsParsing {
     
     /// The modified time for the `Folder`, converted to a `Date` type
     @objc public private(set) var modifiedDate: Date?
```
Comment (bryantjustin) [inline Sources/Shared/Models/Folder.swift]: `modifiedDate` and `createdDate` can probably be removed.

File: Sources/Shared/Models/Folder.swift [hunk 3]
```diff
@@ -100,15 +110,26 @@ public class Folder: VIMModelObject, ConnectionsProviding, ConnectionsParsing {
     public override func getClassForObjectKey(_ key: String!) -> AnyClass? {
         return Mappings.classesByEncodingKeys[key]
     }
+
+    private func formatDate(from dateString: String?) -> Date? {
```
Comment (bryantjustin) [inline Sources/Shared/Models/Folder.swift]: You can remove this. In the future if you're copying something like this, you should really consider refactoring the method into a protocol as opposed to copying and pasting.

File: Sources/Shared/Models/UserItem.swift [hunk 4]
```diff
@@ -26,22 +26,45 @@
...
 
 extension UserItem {
-    
     public enum UserItemType: String {
         case folder
         case video
     }
 }
+
+extension UserItem {
```
Comment (bryantjustin) [inline Sources/Shared/Models/UserItem.swift]: This one can be `private`, or combined with the previous extension and make the struct `private`

**Review by bryantjustin — CHANGES_REQUESTED** [review]
As per our conversation, we shouldn't have to convert between ISO8601 string to `NSDate`. All those properties should be changed.

**Review by bryantjustin — APPROVED** [review]
👍

**Pull request merged**